fhir/scripts/populate.js

```diff
@@ -0,0 +1,149 @@
+
+	// Fail if ther is an error
+	if (err) {
+		console.error(err);
+		process.exit(err);
+	}
+
+	// Connect to our Database
+	let db = client.db(mongoConfig.db_name);
+	let collectionList = await db.listCollections().toArray();
+	let currentCollections = collectionList.map(coll => coll.name);
+
+	// Iterate over our profiles and insert  all of our documents
+	let profile_names = Object.keys(profiles);
+	for (let name of profile_names) {
+		let documents = profiles[name];
+
+		// Add a validator for the schemas
+		// let [ schemaErr, collection ] = await asyncHandler(db.createCollection(name, {
+		// 	validator: { $jsonSchema: SCHEMA_MAP[name] }
+		// }));
+    //
+		// // Skip this iteration if there was an error creating the schema
+		// if (schemaErr) {
+		// 	console.warn(`Unable to create collection for ${name} profile. Skipping.`);
+		// 	console.error(schemaErr);
+		// 	return;
+		// }
+		let collection = db.collection(name);
+
+		if (reset && currentCollections.indexOf(name) > -1) {
+			await asyncHandler(collection.drop());
+		}
+
+		// Insert all of our documents
+		console.log(`\nInserting documents for the ${name} profile.`);
+		let [insertErr, results] = await asyncHandler(collection.insertMany(documents));
+
+		if (insertErr) {
+			console.warn(`Unable to insert documents for ${name} profile.`);
+			console.error(insertErr);
+		}
+		else {
+			console.log(`Success. Inserted ${results.result.n} documents in ${name} profile.`);
+		}
+	}
+
+	// Close our connection
+	console.log('\nClosing connection\n');
+	client.close();
+};
+
+/**
+ * @function loadDocumentsForProfile
+ * @summary Search the fixtures directory and find matching files
+ */
+let loadDocumentsForProfile = profile_name => {
+	// Get a list of json files in fixtures
+	let basePath = path.join(__dirname, '../fixtures');
+	let profiles = glob.sync(path.join(basePath, '**/*.json'));
+	let modules = profiles
+		.map(profile_path => require(profile_path))
+		.filter(module => module.resourceType === profile_name);
+
+	return modules;
+};
+
+/**
+ * @function parseArgs
+ * @summary Parse profiles from commmand line that we want to seed mongo with
+ * @TODO: We should use something like commander later to make this cleaner
+ * and force only profiles that we know are valid. Maybe add a flag for
+ * clearing current collection first
+ */
+let parseArgs = () => {
+	let { version } = package.version;
+
+	// Setup the CLI and parse arguments
+	program
+		.version(version)
+		.option('-p, --profiles [profile]', `Comma separated list of profiles. Valid ones are ${VALID_PROFILE_NAMES.join(',')}`)
+		.option('-r, --reset', 'Reset the collection you are insertng documents into.');
+
+	program.on('--help', () => {
+		console.log();
+		console.log('  Examples:\n');
+		console.log('    $ docker-compose exec fhir yarn populate -p \'Patient,Observation\' -r');
+		console.log('    $ node scripts/populate -p \'Patient,Observation\' -r');
```

Comment:
Needing to escape the string looks weird.  Would it be better to allow multiple -p arguments, so it looks more like `-p Patient -p Observation` etc?  I guess even better would be to accept an array of arguments without the flag at all.

`docker-compose exec fhir yarn populate -r Patient Observation`

Reply:
`docker-compose exec fhir yarn populate -p Patient Observation` would not work without more configuration on commander.  It would read Patient as the argument for p and Observation would get treated as an extraneous arg, not related to p.  I could if I parse the args myself outside of commander. You can also do variadic arguments(-p foo -p bar) but I'm not convinced that's any better than passing in the args as a string. Seems like a different way to do the same thing. `-p Patient,Observation` is less verbose and still very readable. Quotes are actually only needed if there are spaces so I can remove that from the example.